core/src/main/java/org/infinispan/interceptors/impl/CacheMgmtInterceptor.java

```diff
@@ -344,12 +359,14 @@ public long getAverageRemoveTime() {
    }
 
    @ManagedAttribute(
-         description = "Number of entries currently in memory including expired entries",
+         description = "Number of entries currently in memory excluding expired entries",
          displayName = "Number of current cache entries",
          displayType = DisplayType.SUMMARY
    )
    public int getNumberOfEntries() {
-      return dataContainer.sizeIncludingExpired();
+      long flags = EnumUtil.bitSetOf(Flag.CACHE_MODE_LOCAL, Flag.SKIP_CACHE_LOAD);
+      SizeCommand command = commandsFactory.buildSizeCommand(flags);
+      return (Integer) invoker.invoke(icf.createInvocationContext(false, UNBOUNDED), command);
```

Comment:
By the way if you just want in memory but excluding expired. You could call `dataContainer.size` as this will not count expired entries.